zlib-rs/src/deflate/hash_calc.rs

```diff
@@ -117,7 +117,7 @@ impl RollHashCalc {
 
             let head = state.head.as_slice()[hm];
             if head != idx {
-                state.prev.as_mut_slice()[idx as usize & state.w_mask] = head;
+                state.prev.as_mut_slice()[idx as usize & state.w_mask()] = head;
```

Comment:
I see some further improvements for level 2 when moving the `state.w_mask()` call out of the loop. Did/can you try that?